src/main/java/Duke.java

```diff
@@ -1,10 +1,149 @@
+            return;
+        }
+        printResponse("Here are the tasks in your list:");
+        for (int i = 0; i < taskCount; ++i) {
+            printResponse(String.format("%d. %s", i + 1, tasks[i]));
+        }
+    }
+
+    private static void markTaskDone(String cmd) {
+        String[] words = cmd.split(" ");
+        if (words.length > 1 && words[1].matches(REGEX_NUM)) {
+            int index = Integer.parseInt(words[1]);
+            if (index <= taskCount && index > 0) {
+                tasks[index - 1].markAsDone();
+
+                printResponse("Ok! I've marked this task as done:");
+                printResponse(String.format("  %s", tasks[index - 1]));
+            } else System.out.printf("Error: Task %d does not exist", index);
+        } else {
+            printResponse("Usage: done [task index]");
+            printResponse("Example: done 1");
+        }
+    }
+
+    private static Todo parseTodoCommand(String cmd) {
+        int descriptionIndex = TODO_COMMAND.length();
+        String description = cmd.substring(descriptionIndex);
+        return new Todo(description);
+    }
+
+    private static Deadline parseDeadlineCommand(String cmd) {
+        int descriptionIndex = DEADLINE_COMMAND.length();
+        int dateIndex = cmd.indexOf(BY_ARGUMENT);
+        String description = cmd.substring(descriptionIndex, dateIndex - 1);
+        String date = cmd.substring(dateIndex + BY_ARGUMENT.length());
+        return new Deadline(description, date);
+    }
+
+    private static Event parseEventCommand(String cmd) {
+        int descriptionIndex = EVENT_COMMAND.length();
+        int dateIndex = cmd.indexOf(AT_ARGUMENT);
+        String description = cmd.substring(descriptionIndex, dateIndex - 1);
+        String date = cmd.substring(dateIndex + AT_ARGUMENT.length());
+        return new Event(description, date);
+    }
+
+    private static Task parseTaskCommand(String cmd) {
+        if (cmd.startsWith(TODO_COMMAND)) {
+            return parseTodoCommand(cmd);
+        } else if (cmd.startsWith(DEADLINE_COMMAND)) {
+            return parseDeadlineCommand(cmd);
+        } else if (cmd.startsWith(EVENT_COMMAND)) {
+            return parseEventCommand(cmd);
+        }
+        return new Task(cmd);
+    }
+
+    private static void addTask(Task task) {
+        printResponse(String.format("Added: %s", task));
+        tasks[taskCount++] = task;
+        printResponse(
+                String.format(
+                        "You now have %d task%s in your list.",
+                        taskCount, (taskCount == 1 ? "" : "s")));
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String logo =
```

Comment:
Perhaps it would be better to put your welcome message and the logo into a method of its own

Reply:
Done